src/app/routes/getInitialData/index.js

```diff
@@ -18,7 +20,47 @@ const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+export const validateRendererEnvironment = environment => {
+  const validRendererEnvironments = ['test', 'live'];
+
+  if (
+    !(
+      environment &&
+      validRendererEnvironments.includes(environment.toLowerCase())
+    )
+  ) {
+    logger.warn(
+      `Invalid parameter value [${environment}]. Usage: renderer_env=${validRendererEnvironments.join(
+        '|',
+      )}`,
+    );
+  }
+};
+
+const getPathFromUrl = url => {
+  return url.split(/[?]/)[0];
+};
+
+const checkAndAppendParams = pathname => {
+  const url = getPathFromUrl(pathname);
+  const params = qs.parse(pathname.substring(url.length));
+
+  if (params && params.renderer_env) {
+    params.renderer_env = params.renderer_env.toLowerCase();
+    validateRendererEnvironment(params.renderer_env);
+  }
+
+  return `${baseUrl}${url.replace(ampRegex, '')}.json${qs.stringify(
```

Comment:
`${baseUrl}${url` doesn't read well. Should be `${baseUrl}${pathname` like it originally was